File: src/main/java/com/qtech/bubbles/graphics/Border.java
package com.qtech.bubbles.graphics;

import com.qtech.bubbles.common.GraphicsProcessor;

import java.awt.*;

public class Border {
    private final Insets borderInsets;
    private Paint paint = new Color(0, 0, 0);

    public Border(Insets insets) {
        this.borderInsets = insets;
    }

    public Border(int top, int left, int bottom, int right) {
        this.borderInsets = new Insets(top, left, bottom, right);
    }

    public void paintBorder(GraphicsProcessor gp, int x, int y, int width, int height) {
        Insets insets = getBorderInsets();

        //  Set paint.
        gp.setPaint(paint);

        //  Draw rectangles around the component, but do not draw
        //  in the component area itself.
        gp.fillRect(x + insets.left, y, width - insets.left - insets.right, insets.top);
        gp.fillRect(x, y, insets.left, height);
        gp.fillRect(x + width - insets.right, y, insets.right, height);
        gp.fillRect(x + insets.left, y + height - insets.bottom, width - insets.left - insets.right, insets.bottom);
    }

    public Paint getPaint() {
        return paint;
    }

    public void setPaint(Paint paint) {
        this.paint = paint;
    }

    public Insets getBorderInsets() {
        return borderInsets;
    }

    public void setBorderTop(int topWidth) {
        borderInsets.top = topWidth;
    }

    public void setBorderLeft(int leftWidth) {
        borderInsets.left = leftWidth;
    }

    public void setBorderBottom(int bottomWidth) {
        borderInsets.bottom = bottomWidth;
    }

    public void setBorderRight(int rightWidth) {
        borderInsets.right = rightWidth;
    }

    public int getBorderTop() {
        return borderInsets.top;
    }

    public int getBorderLeft() {
        return borderInsets.left;
    }

    public int getBorderBottom() {
        return borderInsets.bottom;
    }

    public int getBorderRight() {
        return borderInsets.right;
    }
}

File: src/main/kotlin/com/qtech/bubbles/graphics/Border.kt
package com.qtech.bubbles.graphics

import java.awt.Color
import java.awt.Paint

open class Border {
    val borderInsets: Insets
    var isBorderOpaque = false
    var paint: Paint = Color(0, 0, 0)

    constructor(insets: Insets) {
        borderInsets = insets
    }

    constructor(top: Int, left: Int, bottom: Int, right: Int) {
        borderInsets = Insets(top, left, bottom, right)
    }

    open fun paintBorder(g: GraphicsProcessor, x: Int, y: Int, width: Int, height: Int) {
        val insets = borderInsets

        //  Set paint.
        g.paint(paint)

        //  Draw rectangles around the component, but do not draw
        //  in the component area itself.
        g.rectF(x + insets.left, y, width - insets.left - insets.right, insets.top)
        g.rectF(x, y, insets.left, height)
        g.rectF(x + width - insets.right, y, insets.right, height)
        g.rectF(x + insets.left, y + height - insets.bottom, width - insets.left - insets.right, insets.bottom)
    }

    var borderTop: Int
        get() = borderInsets.top
        set(topWidth) {
            borderInsets.top = topWidth
        }
    var borderLeft: Int
        get() = borderInsets.left
        set(leftWidth) {
            borderInsets.left = leftWidth
        }
    var borderBottom: Int
        get() = borderInsets.bottom
        set(bottomWidth) {
            borderInsets.bottom = bottomWidth
        }
    var borderRight: Int
        get() = borderInsets.right
        set(rightWidth) {
            borderInsets.right = rightWidth
        }
}